### PR TITLE
MM-10863: Handle non-API errors with redirect to webapp

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
The /login/sso/saml was returning to the browser json errors. Now redirect to an error page in the webapp.

#### Ticket Link
[MM-10863](https://mattermost.atlassian.net/browse/MM-10863)

### CODE REVIEW

File: web/handlers.go [hunk 1]
```diff
@@ -157,8 +157,12 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			c.Err.IsOAuth = false
 		}
 
-		w.WriteHeader(c.Err.StatusCode)
-		w.Write([]byte(c.Err.ToJson()))
+		if IsApiCall(r) {
```
Comment (enahum) [inline web/handlers.go]: Jesús I don't think this will work as this endpoint will never be an API call, now the one thing we need to make sure is that if SAML returns an error and it does not have the action equals mobile like https://github.com/mattermost/mattermost-server/blob/master/web/saml.go#L90 then it should redirect, otherwise it should return the json object

Comment (enahum) [reply web/handlers.go]: Ok @jespino as I said earlier this won't work.. I've been exploring options and have a solution basically we can check for the existence of a header like `X-Mobile-App` that if set then we return the json object if not then we redirect to the error page.

In case of the previous versions of the mobile app it will render the error page as it always have and with the new version of the app then it will render the error by parsing the json object.

basically your block would change to this:
```go
if len(r.Header.Get("X-Mobile-App")) > 0 {
    w.WriteHeader(c.Err.StatusCode)
    w.Write([]byte(c.Err.ToJson()))
} else {
    utils.RenderWebAppError(w, r, c.Err, c.App.AsymmetricSigningKey())
}
```

Comment (jespino) [reply web/handlers.go]: I think we still need the `IsApiCall(r)` check because we want to return redirects only on non-api errors.

Comment (jespino) [reply web/handlers.go]: I tested it manually, but I want to add unit tests.

**Comment by jespino** [comment]
Added unit tests for this specific case

**Comment by enahum** [comment]
@jespino build failing

**Comment by jespino** [comment]
fixed
